components/compliance-service/profiles/db/store.go

```diff
@@ -256,33 +256,49 @@ func (s *Store) ReadMarketTar(name string, version string) ([]byte, error) {
 	return tarBlob, nil
 }
 
-func (s *Store) latestVersion(namespace string, name string) (*inspec.Metadata, error) {
-	sort := "name"
-	order := "ASC"
-	metadataList, err := s.ListProfilesMetadata(namespace, name, sort, order)
+func (s *Store) latestVersion(namespace string, name string) (string, error) {
+	query := `
+		SELECT info->>'version'
+		FROM store_profiles WHERE
+		sha256 IN (SELECT sha256 FROM store_namespace WHERE owner=$1)
+		AND info->>'name' = $2
+	`
+
+	rows, err := s.DB.Query(query, namespace, name)
 	if err != nil {
-		return nil, err
+		return "", err
 	}
-	var latest inspec.Metadata
-	for _, metadata := range metadataList {
-		v1, _ := semver.Make(latest.Version)
-		v2, _ := semver.Make(metadata.Version)
-		if metadata.Name == name && v2.GT(v1) {
-			latest = metadata
+	defer rows.Close() // nolint: errcheck
+
+	versions := make([]semver.Version, 0)
+
+	for rows.Next() {
+		var version semver.Version
+		err := rows.Scan(&version)
+		if err != nil {
+			return "", err
 		}
+		versions = append(versions, version)
+	}
+	err = rows.Err()
+	if err != nil {
+		return "", err
 	}
 
-	return &latest, nil
+	semver.Sort(versions)
```

Comment:
Interesting.  I suppose this is also a behavior change/bug fix since previously we would have just gotten whatever the lexicographic sort order of the versions was?

Reply:
This function was doing an equivalent thing but more verbosely: https://github.com/chef/automate/blob/5651c419def4fd1fb56ec67c19509589d718b243/components/compliance-service/profiles/db/store.go#L266-L273

We do sort by the version string in the database in other places though.

Reply:
Aha! Missed that bit, thanks!